src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Apprenticeship.cs

```diff
@@ -36,12 +39,43 @@ public ApprenticeshipStatus Status
             }
         }
 
-        
+        public DateTime? CompletionDate { get; set; }
 
         public Apprenticeship()
         {
             DataLockStatus = new List<DataLockStatus>();
             PriceHistory = new List<PriceHistory>();
         }
+
+        public virtual void Complete(DateTime completionDate)
+        {
+            if (Status != ApprenticeshipStatus.Live)
+            {
+                throw new InvalidOperationException("Apprenticeship has to be live to be completed");
+            }
+
+            StartTrackingSession(UserAction.CompletionPayment, Party.None, Cohort.EmployerAccountId, Cohort.ProviderId, null);
+            ChangeTrackingSession.TrackUpdate(this);
+            PaymentStatus = PaymentStatus.Completed;
+            CompletionDate = completionDate;
+            ChangeTrackingSession.CompleteTrackingSession();
+
+            Publish(() => new ApprenticeshipCompletedEvent{ ApprenticeshipId = Id, CompletionDate = completionDate});
+        }
+
+        public virtual void UpdateCompletionDate(DateTime completionDate)
+        {
+            if (Status != ApprenticeshipStatus.Completed)
```

Comment:
As above, we have a PaymentStatus property we can use

Reply:
Easier to use the payment status here as it's a one to one mapping

Reply:
If we are keeping the ApprenticeStatus field for the moment. Does it make sense to use it here as well as in the Completed metod.